src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelPage/AddPersonnelSideSheet/index.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { ModalSideSheet, Button, Spinner } from '@equinor/fusion-components';
+import Personnel from '../../../../../../../models/Personnel';
+import Person from '../../../../../../../models/Person';
+import * as uuid from "uuid/v1";
+import * as classNames from 'classnames'
+import * as styles from './styles.less'
+import { useComponentDisplayClassNames, useCurrentContext } from '@equinor/fusion';
+import { generateRowTemplate, generateColumnTemplate } from './utils';
+import Header from './AddPersonnelFormHeader';
+import { useAppContext } from '../../../../../../../appContext';
+import { useContractContext } from '../../../../../../../contractContex';
+import AddPersonnelFormTextInput from './AddPersonnelFormTextInput';
+import useAddPersonnelForm from '../hooks/useAddPersonnelForm';
+
+
+type AddPersonnelToSideSheetProps = {
+  isOpen: boolean;
+  selectedPersonnel:Personnel[] | null
+  setIsOpen: (state:boolean) => void;
+}
+
+
+const AddPersonnelSideSheet: React.FC<AddPersonnelToSideSheetProps> = ({isOpen,setIsOpen,selectedPersonnel}) => {
+  const currentContext = useCurrentContext()
+  const currentContract = useContractContext()
+  const {
+    formState,
+    setFormState,
+    resetForm,
+    isFormValid,
+    isFormDirty,
+  } =  useAddPersonnelForm(selectedPersonnel)
+  const { apiClient } = useAppContext();
+  
+  const [saveInProgress,setSaveInProgress] = React.useState<Boolean>(false);
+  
+  const  savePersonnelChangesAsync = async () => {
+    const contractId = currentContract.contract?.id;
+
+
+    if(!currentContext?.id || !contractId )
+      return  
+    
+    setSaveInProgress(true);
+    await Promise.all(formState
+      .map(async person => await apiClient.updatePersonnelAsync(currentContext.id,contractId,person)))
+      .then(() => {
+        setSaveInProgress(false)
+        setIsOpen(false);
+      })
+      .catch(() => setSaveInProgress(false))  
```

Comment:
Should probably let the use know an error occurred and the personnel might not have been saved